ui/app/pages/confirm-transaction-base/confirm-transaction-base.component.js

```diff
@@ -251,6 +263,37 @@ export default class ConfirmTransactionBase extends Component {
               primaryValueTextColor="#2f9ae0"
             />
           </div>
+          {useNonceField ? <div>
+            <div className="confirm-detail-row">
+              <div className="confirm-detail-row__label">
+                { this.context.t('nonceFieldHeading') }
+              </div>
+              <div className="custom-nonce-input">
+                <TextField
+                  type="number"
+                  min="0"
+                  placeholder={ this.context.t('nonceFieldPlaceholder') }
+                  onChange={({ target: { value } }) => {
+                    if (!value.length || Number(value) < 0) {
+                      updateCustomNonce('')
+                    } else {
+                      const newCustomNonce = Math.floor(value)
+                      if (newCustomNonce > nextNonce) {
+                        this.setState({ submitWarning: this.context.t('nextNonceWarning') })
+                      } else {
+                        this.setState({ submitWarning: '' })
+                      }
+                      updateCustomNonce(String(newCustomNonce))
+                      getNextNonce()
```

Comment:
If this results in the nextNonce changing to one that would invalidate the custom nonce, it won't result in a warning. I think this case could be caught by re-validating the custom nonce field in the lifecycle function when the `nextNonce` prop changes